### PR TITLE
[cli] [servenv] Migrate flags used by grpc servers to `pflag`

### PR DESCRIPTION
**Pull request by ajm188**
## Description

This migrates the flags used by the grpc servers calling into `servenv.Init`, `servenv.Run`, and `servenv.RunDefault`.

⚠️ it also removes the deprecated profliing flags, which were deprecated approximately 18 months ago ⚠️ 

## Related Issue(s)

- #11146 
- #11144 

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

File: go/vt/servenv/grpc_server.go [hunk 1]
```diff
@@ -84,37 +64,105 @@ var (
...
+	gRPCEnableOptionalTLS bool
+	// gRPCServerCA if specified will combine server cert and server CA.
+	gRPCServerCA string
 )
 
+// RegisterGRPCServerFlags registers flags required to run a gRPC server via Run
+// or RunDefault.
+//
+// `go/cmd/*`` entrypoints should call this function before
+// ParseFlags(WithArgs)? if they wish to expose run a gRPC server.
```
Comment (deepthi) [inline go/vt/servenv/grpc_server.go:123]: expose or run but not both?

**Review by deepthi — COMMENTED** [review]
Mostly LGTM. We do need to document the deleted flags in the release notes, so I added that label.

File: go/vt/servenv/grpc_server.go (hunk 1 above)
Comment (ajm188) [reply go/vt/servenv/grpc_server.go:123]: yeah just "run", bad copy-pasta

File: doc/releasenotes/15_0_0_summary.md [hunk 2]
```diff
@@ -17,6 +17,8 @@
 
 ### Breaking Changes
 
+- The deprecated `--cpu_profile` flag has been removed. Please use the `--pprof` flag instead.
```
Comment (deepthi) [inline doc/releasenotes/15_0_0_summary.md:20]: Two other flags were deleted. Can you please document those and then merge?
```
       --mem-profile-rate int                                             deprecated: use '-pprof=mem' instead (default 524288)
       --mutex-profile-fraction int                                       deprecated: use '-pprof=mutex' instead
```

**Review by deepthi — APPROVED** [review]
May merge once the docs are complete.

File: doc/releasenotes/15_0_0_summary.md (hunk 2 above)
Comment (ajm188) [reply doc/releasenotes/15_0_0_summary.md:20]: oh whoops, yep!!

**Pull request merged**